### PR TITLE
Allow to disable generating documentation for functions with `#private` notation

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
Closes: https://github.com/metarhia/metadoc/issues/58

### CODE REVIEW

**Comment by belochub** [comment]
@SemenchenkoVitaliy What about cases where the user wants to avoid having specific functions/methods that start with an underscore or some other character in the documentation? Or those that do not start with some kind of special character, but still are not supposed to be called by the external users?

**Comment by SemenchenkoVitaliy** [comment]
@belochub In this case, we should have some kind of filter which will be in the config file. e.g.
```json
{
  "filteredFunctions": [
    "path.to.ClassName.prototype.privateMethodName",
    "unexportedFunction",
  ]
}
```
Or we can allow to specify `RegExp`s instead of/alongside with methods paths though I'm not a fun of this idea. WDYT?

**Comment by nechaido** [comment]
I do not like the idea to specify omitted functions in the config. I'd prefer to have some kind of `#private` notation to state that function\property is private.
Example:
```js
// #private
// Documentation for this function will not be generated unless it is specified
// via cli or config file
const _undescoreFunction = (a, b, c) => {};
```

**Comment by belochub** [comment]
@SemenchenkoVitaliy I don't think it should be specified in configs, it'd be better to add some kind of special syntax in the function's/method's comments. This is up for discussion though.

File: bin/cli.js [hunk 1]
```diff
@@ -26,6 +26,10 @@ const args = yargs
     type: 'string',
     describe: 'file with footer for the resulting doc file',
   })
+  .option('gen-private-methods', {
```
Comment (belochub) [inline bin/cli.js]: Maybe just call it `--private`?

File: test/example.md [hunk 2]
```diff
@@ -207,9 +207,10 @@ ExampleClass description
 
 - [`<string>`][string] setter for `prop1`
 
-### ExampleClass.method2(num)
+### ExampleClass.method2(str, \_num)
```
Comment (lundibundi) [inline test/example.md]: Is this still relevant?

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,13 @@ and this project adheres to
 
 ## [Unreleased][unreleased]
 
+### Added
+
+- Allow to disable generating documentation for functions with `#private`
+  notation. Documentation for such functions will not be generated by default.
```
Comment (lundibundi) [inline CHANGELOG.md]:
```suggestion
  header in the documentation comment. Documentation for such functions will not be generated by default.
```
?

**Comment by lundibundi** [comment]
Landed in Landed in c0c42058d4e474a3ed1a4580a9eca7200455f99c.

**Pull request closed without merging**